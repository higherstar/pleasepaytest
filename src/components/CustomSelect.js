import React, {Component} from "react";
import Select from "react-select";
import {inject, observer} from "mobx-react";

import "react-select/dist/react-select.css";

@inject("store")
@observer
class CustomSelect extends Component {
    constructor(props) {
        super(props);

        this.store = this.props.store.appState;

        this.state = {
            country: 'AU',
            disabled: false,
            searchable: this.props.searchable,
            clearable: this.props.clearable,
            rtl: false,
        };

        this.updateValue = this.updateValue.bind(this);
    }

    updateValue(newValue) {
        if (this.props.label === 'Country') {
            this.store.selectCountry(newValue);
        } else {
            this.store.selectCurrency(newValue);
        }
    }

    render() {
        let options = [];
        let value = '';
        let {label} = this.props;
        if (label === 'Country') {
            options = this.store.countries;
            value = this.store.selectedCountry;
        } else if (label === 'Currency') {
            options = this.store.currencies;
            value = this.store.selectedCurrency;
        }

        return (
            <div className="selector">
                <div className="label-area">
                    <span>{label}</span>
                </div>
                <div className="select-area">
                    <Select
                        id="state-select"
                        name="selected-state"
                        onBlurResetsInput={false}
                        onSelectResetsInput={false}
                        autoFocus
                        options={options}
                        simpleValue
                        disabled={this.state.disabled}
                        value={value}
                        onChange={this.updateValue}
                        rtl={this.state.rtl}
                        searchable={this.state.searchable}
                        clearable={this.state.clearable}
                    />
                </div>
            </div>
        );
    }
}

export default CustomSelect;