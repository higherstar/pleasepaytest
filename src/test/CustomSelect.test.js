import React from 'react';
import {shallow} from 'enzyme';

import CustomSelect from '../../src/components/CustomSelect';

describe('CustomSelect component', () => {
    let component;
    let store;

    beforeEach(() => {
        store = {
            countries: [
                {currency: "SEK", id: "58f9f682d86a9551bbffeaa8", label: "Sweden", value: "sweden"},
                {currency: "NZD", id: "58f9f682d86a9551bbffeab6", label: "New Zealand", value: "new-zealand"},
                {currency: "EUR", id: "58f9f682d86a9551bbffea9d", label: "Netherlands", value: "netherlands"},
                {currency: "EUR", id: "58f9f682d86a9551bbffea92", label: "Austria", value: "austria"},
            ],
            currencies: [
                {id: "58f9f682d86a9551bbffeadb", value: "MXN", label: "MXN"},
                {id: "58f9f682d86a9551bbffeadc", value: "NZD", label: "NZD"},
                {id: "58f9f682d86a9551bbffeae3", value: "RUB", label: "RUB"},
                {id: "58f9f682d86a9551bbffead6", value: "AUD", label: "AUD"}
            ],
            selectedCountry: "new-zealand",
            selectedCurrency: "NZD"
        };
        component = shallow(
            <CustomSelect
                store={store}
                label="Country"
                searchable={true}
                clearable={true} />
        );
    });

    test('Render', () => {
        expect(component).toMatchSnapshot();
    });


    test('render Markdown in preview mode', () => {
        const wrapper = shallow(
            <CustomSelect
                store={store}
                label="Country"
                searchable={true}
                clearable={true} />
        );

        expect(wrapper).toMatchSnapshot();

        wrapper.find('[class="Select-control"]').simulate('click');

        expect(wrapper).toMatchSnapshot();
    });

})

