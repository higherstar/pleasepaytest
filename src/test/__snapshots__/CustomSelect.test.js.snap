// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomSelect component Render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <inject-CustomSelect-with-store
    clearable={true}
    label="Country"
    searchable={true}
    store={
      Object {
        "countries": Array [
          Object {
            "currency": "SEK",
            "id": "58f9f682d86a9551bbffeaa8",
            "label": "Sweden",
            "value": "sweden",
          },
          Object {
            "currency": "NZD",
            "id": "58f9f682d86a9551bbffeab6",
            "label": "New Zealand",
            "value": "new-zealand",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea9d",
            "label": "Netherlands",
            "value": "netherlands",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea92",
            "label": "Austria",
            "value": "austria",
          },
        ],
        "currencies": Array [
          Object {
            "id": "58f9f682d86a9551bbffeadb",
            "label": "MXN",
            "value": "MXN",
          },
          Object {
            "id": "58f9f682d86a9551bbffeadc",
            "label": "NZD",
            "value": "NZD",
          },
          Object {
            "id": "58f9f682d86a9551bbffeae3",
            "label": "RUB",
            "value": "RUB",
          },
          Object {
            "id": "58f9f682d86a9551bbffead6",
            "label": "AUD",
            "value": "AUD",
          },
        ],
        "selectedCountry": "new-zealand",
        "selectedCurrency": "NZD",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "clearable": true,
      "label": "Country",
      "searchable": true,
      "store": Object {
        "countries": Array [
          Object {
            "currency": "SEK",
            "id": "58f9f682d86a9551bbffeaa8",
            "label": "Sweden",
            "value": "sweden",
          },
          Object {
            "currency": "NZD",
            "id": "58f9f682d86a9551bbffeab6",
            "label": "New Zealand",
            "value": "new-zealand",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea9d",
            "label": "Netherlands",
            "value": "netherlands",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea92",
            "label": "Austria",
            "value": "austria",
          },
        ],
        "currencies": Array [
          Object {
            "id": "58f9f682d86a9551bbffeadb",
            "label": "MXN",
            "value": "MXN",
          },
          Object {
            "id": "58f9f682d86a9551bbffeadc",
            "label": "NZD",
            "value": "NZD",
          },
          Object {
            "id": "58f9f682d86a9551bbffeae3",
            "label": "RUB",
            "value": "RUB",
          },
          Object {
            "id": "58f9f682d86a9551bbffead6",
            "label": "AUD",
            "value": "AUD",
          },
        ],
        "selectedCountry": "new-zealand",
        "selectedCurrency": "NZD",
      },
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "clearable": true,
        "label": "Country",
        "searchable": true,
        "store": Object {
          "countries": Array [
            Object {
              "currency": "SEK",
              "id": "58f9f682d86a9551bbffeaa8",
              "label": "Sweden",
              "value": "sweden",
            },
            Object {
              "currency": "NZD",
              "id": "58f9f682d86a9551bbffeab6",
              "label": "New Zealand",
              "value": "new-zealand",
            },
            Object {
              "currency": "EUR",
              "id": "58f9f682d86a9551bbffea9d",
              "label": "Netherlands",
              "value": "netherlands",
            },
            Object {
              "currency": "EUR",
              "id": "58f9f682d86a9551bbffea92",
              "label": "Austria",
              "value": "austria",
            },
          ],
          "currencies": Array [
            Object {
              "id": "58f9f682d86a9551bbffeadb",
              "label": "MXN",
              "value": "MXN",
            },
            Object {
              "id": "58f9f682d86a9551bbffeadc",
              "label": "NZD",
              "value": "NZD",
            },
            Object {
              "id": "58f9f682d86a9551bbffeae3",
              "label": "RUB",
              "value": "RUB",
            },
            Object {
              "id": "58f9f682d86a9551bbffead6",
              "label": "AUD",
              "value": "AUD",
            },
          ],
          "selectedCountry": "new-zealand",
          "selectedCurrency": "NZD",
        },
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`CustomSelect component render Markdown in preview mode 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <inject-CustomSelect-with-store
    clearable={true}
    label="Country"
    searchable={true}
    store={
      Object {
        "countries": Array [
          Object {
            "currency": "SEK",
            "id": "58f9f682d86a9551bbffeaa8",
            "label": "Sweden",
            "value": "sweden",
          },
          Object {
            "currency": "NZD",
            "id": "58f9f682d86a9551bbffeab6",
            "label": "New Zealand",
            "value": "new-zealand",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea9d",
            "label": "Netherlands",
            "value": "netherlands",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea92",
            "label": "Austria",
            "value": "austria",
          },
        ],
        "currencies": Array [
          Object {
            "id": "58f9f682d86a9551bbffeadb",
            "label": "MXN",
            "value": "MXN",
          },
          Object {
            "id": "58f9f682d86a9551bbffeadc",
            "label": "NZD",
            "value": "NZD",
          },
          Object {
            "id": "58f9f682d86a9551bbffeae3",
            "label": "RUB",
            "value": "RUB",
          },
          Object {
            "id": "58f9f682d86a9551bbffead6",
            "label": "AUD",
            "value": "AUD",
          },
        ],
        "selectedCountry": "new-zealand",
        "selectedCurrency": "NZD",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "clearable": true,
      "label": "Country",
      "searchable": true,
      "store": Object {
        "countries": Array [
          Object {
            "currency": "SEK",
            "id": "58f9f682d86a9551bbffeaa8",
            "label": "Sweden",
            "value": "sweden",
          },
          Object {
            "currency": "NZD",
            "id": "58f9f682d86a9551bbffeab6",
            "label": "New Zealand",
            "value": "new-zealand",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea9d",
            "label": "Netherlands",
            "value": "netherlands",
          },
          Object {
            "currency": "EUR",
            "id": "58f9f682d86a9551bbffea92",
            "label": "Austria",
            "value": "austria",
          },
        ],
        "currencies": Array [
          Object {
            "id": "58f9f682d86a9551bbffeadb",
            "label": "MXN",
            "value": "MXN",
          },
          Object {
            "id": "58f9f682d86a9551bbffeadc",
            "label": "NZD",
            "value": "NZD",
          },
          Object {
            "id": "58f9f682d86a9551bbffeae3",
            "label": "RUB",
            "value": "RUB",
          },
          Object {
            "id": "58f9f682d86a9551bbffead6",
            "label": "AUD",
            "value": "AUD",
          },
        ],
        "selectedCountry": "new-zealand",
        "selectedCurrency": "NZD",
      },
    },
    "ref": [Function],
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "clearable": true,
        "label": "Country",
        "searchable": true,
        "store": Object {
          "countries": Array [
            Object {
              "currency": "SEK",
              "id": "58f9f682d86a9551bbffeaa8",
              "label": "Sweden",
              "value": "sweden",
            },
            Object {
              "currency": "NZD",
              "id": "58f9f682d86a9551bbffeab6",
              "label": "New Zealand",
              "value": "new-zealand",
            },
            Object {
              "currency": "EUR",
              "id": "58f9f682d86a9551bbffea9d",
              "label": "Netherlands",
              "value": "netherlands",
            },
            Object {
              "currency": "EUR",
              "id": "58f9f682d86a9551bbffea92",
              "label": "Austria",
              "value": "austria",
            },
          ],
          "currencies": Array [
            Object {
              "id": "58f9f682d86a9551bbffeadb",
              "label": "MXN",
              "value": "MXN",
            },
            Object {
              "id": "58f9f682d86a9551bbffeadc",
              "label": "NZD",
              "value": "NZD",
            },
            Object {
              "id": "58f9f682d86a9551bbffeae3",
              "label": "RUB",
              "value": "RUB",
            },
            Object {
              "id": "58f9f682d86a9551bbffead6",
              "label": "AUD",
              "value": "AUD",
            },
          ],
          "selectedCountry": "new-zealand",
          "selectedCurrency": "NZD",
        },
      },
      "ref": [Function],
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
